src/tools/tool-options.h

```diff
@@ -225,6 +228,10 @@ struct ToolOptions : public Options {
   }
 
   void applyOptionsBeforeParse(Module& module) const {
+    if (enabledFeatures & FeatureSet::BulkMemory) {
+      // Enable this subfeature for backward compatibility.
+      module.features.enable(FeatureSet::BulkMemoryOpt);
+    }
```

Comment:
We don't have a precedent for making features depend on other features, but it makes sense here, I think. Should we also disable BulkMemoryOpt when BulkMemory is disabled for symmetry?

Reply:
Funny you should mention the dependence, I just had to fix a bug that I introduced in a late refactoring, and it's here. PTAL the current version; I think `applyOptionsBeforeParse` isn't actually the right place to do this (because it's const and we can't actually change `enabledFeatures`). But fixing that meant making `parse` virtual so that there would be a place that made sense.
Currently we maintain the invariant that we assert in `hasBulkMemoryOpt`, that bulk-memory-opt is always on when build-memory is on. It makes sense to allow bulk-memory-opt to be on with bulk-memory off though. Although for command-line compatibility, maybe there's an expectation that --disable-bulk-memory should also diable --bulk-memory-opt? (which would presumably matter only when they become default)?